source/ai-server-logger.js

```diff
@@ -0,0 +1,122 @@
+(function() {
+    'use strict';
+
+    angular.module('ai.public.serverLogger', ['ng'])
+        .config(function($provide) {
+            $provide.decorator('$log', function($delegate, $injector) {
+                var logger = $injector.get('AiServerLogger');
+                var delegateError = $delegate.error;
+                var delegateWarn = $delegate.warn;
+                var delegateInfo = $delegate.info;
+                var delegateDebug = $delegate.debug;
+
+                var enhanceLoggingFunction = function(loggingFunction, severity) {
+                    return function() {
+                        var args = [].slice.call(arguments);
```

Comment:
Good point.  I haven't looked at performance of argument slicing. I'll use toArray.  The reason I want it to be an array is because of the loggingFunction.apply which expects an array and JSON stringify will serialize the arguments just fine.